File: cgsv/mrs_2.0.0/mrs/sample/cpp/source/base_loop.cpp
#include <mrs.hpp>

int main( int argc, char** argv ){
    mrs_initialize();
    
    MRS_LOG_DEBUG( "[%s] start", mrs::DateTime::Now().ToString().c_str() );
    for ( int i = 0; i < 3; ++i ){
        mrs_update();
        
        MRS_LOG_DEBUG( "[%s] zzz %d", mrs::DateTime::Now().ToString().c_str(), i + 1 );
        mrs_sleep( 1000 );
    }
    MRS_LOG_DEBUG( "[%s] end", mrs::DateTime::Now().ToString().c_str() );
    
    mrs_finalize();
    return 0;
}

File: cgsv/mrs_2.0.0/mrs/sample/csharp/source/base_loop.cs
using System;

namespace base_loop
{
    class MrsMain : Mrs
    {
        static void Main(string[] args)
        {
			mrs_initialize();

			try
			{
				MRS_LOG_DEBUG("[{0}] start", mrs.DateTime.Now().ToString());
				for (int i = 0; i < 3; ++i)
				{
					mrs_update();
					MRS_LOG_DEBUG("[{0}] zzz {1}", mrs.DateTime.Now().ToString(), i + 1);
					mrs_sleep(1000);
				}
				MRS_LOG_DEBUG("[{0}] end", mrs.DateTime.Now().ToString());
			}
			catch (Exception e)
			{
				Console.WriteLine(e.Message);
			}
			finally
			{
				mrs_finalize();
			}
        }
    }
}
